### ISSUE TITLE
Add all optional kwargs of BaseQuery to the modules

### ISSUE THREAD

**Issue by bsipocz** [astropy/astroquery#1233]
E.g. at least `cache` and `timeout`. It will help to easier debug sporadic serverside issues users run into but we cannot reproduce, see e.g. #1232

**Comment by kakirastern** [comment]
Hi @bsipocz, is it okay for me to work on this issue in preparation for GSoC? Or should I pick something more difficult at this stage?

**Comment by bsipocz** [comment]
No, this is perfectly fine and actually very useful to have!

**Comment by kakirastern** [comment]
Yup, that's awesome. Will start with this then for me to get familiar with the astroquery codebase...

**Comment by bsipocz** [comment]
Getting familiarity with `BaseQuery` will go a long way as the preferred GSoC project will most probably involve improving it.

**Comment by kakirastern** [comment]
Yeah, it is true...

**Comment by kakirastern** [comment]
Note to self: Starting point is here where the `BaseQuery` class is defined at [https://github.com/astropy/astroquery/blob/master/astroquery/query.py#L142](https://github.com/astropy/astroquery/blob/master/astroquery/query.py#L142)

Aim to incorporate initially the optional kwargs `cache` and `timeout` to all modules.

**Comment by kakirastern** [comment]
**1st Batch**

**_Strategy_**: Will commit only one module first in order for reviewer to check the work, then extend to other modules in alphabetical order where possible. Starting with the subset of modules which follow a common API, as listed below:

- [ ] alma/alma
- [ ] atomic/atomic
- [ ] besancon/besancon
- [ ] cds/cds
- [ ] esasky/esasky
- [ ] eso/eso
- [ ] gaia/gaia
- [ ] gama/gama
- [ ] heasarc/heasarc
- [ ] hitran/hitran
- [ ] ibe/ibe
- [ ] irsa/irsa
- [ ] irsa/irsa_dust
- [ ] jplspec/jplspec
- [ ] magpis/magpis
- [ ] mast/mast
- [ ] mpc/mpc
- [ ] nasa_ads/nasa_ads
- [ ] ned/ned
- [ ] nist/nist
- [ ] nrao/nrao
- [ ] nvas/nvas
- [ ] simbad/simbad
- [ ] skyview/skyview
- [ ] splatalogue/splatalogue
- [ ] ukidss/ukidss
- [ ] vamdc/vamdc
- [ ] vizier/vizier
- [ ] vo_conesearch/vo_conesearch
- [ ] vsa/vsa
- [ ] xmatch/xmatch

**Comment by bsipocz** [comment]
@kakirastern - I would suggest to keep the note to self comments in issues and PRs to minimum. You can add them to a personal project board, wiki page, google docs, or any other means to track your work. 
I know it's difficult to find the balance, but given that there are plenty people watching the different repositories as it's quite distractive to receive notifications for these entries.

**Comment by kakirastern** [comment]
@bsipocz Sure, I will try to keep to note to self comments to a minimum.

**Comment by kakirastern** [comment]
Hi, just wanted to double-check that I will need to have added the list of keywords for  `_request` in the BaseQuery class as `self, method, url, params, data, headers, files, save, savedir, timeout, cache, stream, auth, continuation, verify`, and I better use the default values as listed in the code snippet:
```
def _request(self, method, url, params=None, data=None, headers=None,
                 files=None, save=False, savedir='', timeout=None, cache=True,
                 stream=False, auth=None, continuation=True, verify=True): ... 
```
and similar for `_download_file` as in 
```
def _download_file(self, url, local_filepath, timeout=None, auth=None,
                       continuation=True, cache=False, method="GET",
                       head_safe=False, **kwargs): ...
```
And, these are to be applied consistently to all modules. If I have made any mistakes above please correct me. Thanks!

**Comment by kakirastern** [comment]
PR #1419 opened to address this issue.
